EIPS/eip-1850.md

```diff
@@ -0,0 +1,550 @@
+---
+eip: 1850
+title: Hashed Time-Locked Principal Contract transactions
+author: Matthew Black <matthew@atomicloans.io>, Tony Cai <tony@atomicloans.io>, Atomic Loans team <info@atomicloans.io>
+status: Draft
+discussions-to: https://github.com/ethereum/EIPs/issues/1851
+type: Standards Track
+category: ERC
+created: 2019-03-19
+---
+
+## Simple Summary
+
+This EIP describes a script for a generalized debt agreement contract based on Hashed Time-Lock Contract (ERC 1630) transactions according to the Atomic Loans specification (https://arxiv.org/pdf/1901.05117.pdf).
+
+
+## Abstract
+
+A Hashed Time-Locked Principal Contract (HTLPC) is a script that permits a designated party (the "lender") to provide ERC20 denominated funds for a debt agreement for a specified amount of time, with collateral locked on another blockchain (denoted as the collateral blockchain for the purposes of this EIP). It also permits a second party (the "borrower") to withdraw the lent funds once their request to borrow has been approved by the lender, repay the loan after the loan period has passed and in the case of default, liquidate the collateral.
+
+## Motivation
+
+HTLPC transactions are a safe method for exchanging multiple secrets for the purposes of executing a cross-chain debt agreement, due to the ability to recover a percentage of funds from an uncooperative counterparty.
+
+HLTPC's enable cross-chain [atomic loans](https://arxiv.org/pdf/1901.05117.pdf)
```

Comment:
The link to this PDF has been added here for the second time. This would also better fit in the **References** section

Reply:
Great point. Added a reference section.